app/models/display_ad_event.rb

```diff
@@ -10,7 +10,8 @@ class DisplayAdEvent < ApplicationRecord
   VALID_CATEGORIES = [CATEGORY_CLICK, CATEGORY_IMPRESSION].freeze
 
   CONTEXT_TYPE_HOME = "home".freeze
-  VALID_CONTEXT_TYPES = [CONTEXT_TYPE_HOME].freeze
+  CONTEXT_TYPE_ARTICLE = "article".freeze
+  VALID_CONTEXT_TYPES = [CONTEXT_TYPE_HOME, CONTEXT_TYPE_ARTICLE].freeze
```

Comment:
@jaw6 I need your suggestion here.

I have added `CONTEXT_TYPE_ARTICLE` which gets used for display ads which are shown on articles. Earlier we were not using any context for those display ads. With this implementation we have 3-display-ads on home page with 'home' context and two on article page with `article` context.
So first, is this correct?
Why are we not using 5 different context to track the impressions in more depth, like `home-left-1`, `home-left-2`, `home-right`, `article-right` and `article-below-comment`

Reply:
It's a good question! I don't really know _why_, or rather _how_ or even _if_ `context` is used. That attribute has been frozen as `"home"` since [the very beginning](https://github.com/forem/forem/pull/3589/files#diff-a2624b7194585f380b666f690e7c304c61212c39a1f30b397e1c5d1650e27886R214) -- [later PRs](https://github.com/forem/forem/pull/15946/) tweaked the mechanics but we did not change `DisplayAdEvent` even [when we introduced new placement areas](https://github.com/forem/forem/pull/18141), as it wasn't part of [the ask](https://github.com/forem/forem-internal-eng/issues/486) (and I was too :new: here to know `DisplayAdEvent` even existed).

I _think_ a reason that it hasn't been changed is that I don't think it gives us much information beyond what we have _already_ in `DisplayAd#placement_area` -- or, phrased somewhat differently, I think `context` was something @benhalpern assumed "would be" valuable [at the beginning](https://github.com/forem/forem/pull/3589/files#diff-a2624b7194585f380b666f690e7c304c61212c39a1f30b397e1c5d1650e27886R214) and maybe it _isn't_ valuable, but also maybe it isn't dragging us down, so removing it hasn't been a priority, but neither has "enhancing it", so to speak.

But most of that happened well before my time here, I am guessing at a lot here!

I'll defer to @benhalpern and @mirie, my sense is that it probably isn't worth changing in _this_ PR, but we should maybe open a new issue to discuss / prioritize it separately.